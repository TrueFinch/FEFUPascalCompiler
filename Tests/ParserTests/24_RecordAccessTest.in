program Project1;

type 
    a = record
        k: integer;
        l: integer;
    end;
  
var 
    i: a;

begin
    i.k := 0;
    i.l := 4;
    write(i.k);
    write(i.l);
end.