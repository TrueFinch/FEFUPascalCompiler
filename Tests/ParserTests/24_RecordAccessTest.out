                                                          project1                                                           
                                                             ┼                                                               
                                                         MainBlock                                                           
                ┌───────────────────────┬────────────────────┴───────────────────────┐                                       
              type                     var                                   CompoundStatement                               
                ┼                       ┼               ┌────────────────┬───────────┴──────┬────────────────────┐           
            TypeDecl              SimpleVarDecl        :=               :=            FunctionCall         FunctionCall      
 ┌──────────────┴┐                ┌─────┴┐            ┌─┴──────┐       ┌─┴──────┐    ┌──────┴──┐          ┌──────┴──┐        
 a           RecordType           i  SimpleType  RecordAccess  0  RecordAccess  4  write  RecordAccess  write  RecordAccess  
          ┌──────┴───────┐           ┌───┴       ┌──┬─┴           ┌──┬─┴                  ┌──┬─┴               ┌──┬─┴        
          :              :           a           i  k             i  l                    i  k                 i  l          
    ┌─────┴┐       ┌─────┴┐                                                                                                  
    k  SimpleType  l  SimpleType                                                                                             
          ┌┴             ┌┴                                                                                                  
       integer        integer                                                                                                